src/types.rs

```diff
@@ -63,6 +64,46 @@ impl fmt::Display for StateId {
     }
 }
 
+impl FromStr for StateId {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "finalized" => Ok(StateId::Finalized),
+            "justified" => Ok(StateId::Justified),
+            "head" => Ok(StateId::Head),
+            "genesis" => Ok(StateId::Genesis),
+            _ => match s.parse::<Slot>() {
+                Ok(slot) => Ok(Self::Slot(slot)),
+                Err(_) => match try_bytes_from_hex_str(s) {
+                    Ok(root_data) => {
+                        let root = Root::try_from(&root_data[..]).map_err(|err| format!("could not parse state identifier by root from the provided argument {s}: {err}"))?;
```

Comment:
nitpick but I think we can just do this:
```suggestion
                        let root = Root::try_from(&root_data).map_err(|err| format!("could not parse state identifier by root from the provided argument {s}: {err}"))?;
```

or if we can't get the types to line up then I have a slight preference for `root_data.as_ref()` instead of the `&{}[..]` thing

Reply:
I think to do this we would need to implement `From<&Vec<u8>>` on the `Node` type in `ssz_rs`. 
The fallback option `root_data.as_ref()` works without having to touch `ssz_rs`.
```suggestion
                        let root = Root::try_from(root_data.as_ref()).map_err(|err| format!("could not parse state identifier by root from the provided argument {s}: {err}"))?;
```